andela/Ah-backend-aquaman
# #164069234 Search articles by author, tags, favourited and keywords

#### What does this PR do?
- Add search functionality  and custom filtering based on parameters
#### Description of Task to be completed?
- Ensure user can filter articles by author name
- Ensure user can filter articles by tag name
- Ensure user can filter articles by who favourited the articles
#### How should this be manually tested?
- Clone the [repo](https://github.com/andela/Ah-backend-aquaman), cd into `Ah-backend-aquaman` and checkout the branch `ft-search-article-164069234`
    -  pip install -r requirements.txt
    -  python manage.py migrate
    -  python manage.py runserver
    -  create an account, verify your email, login and create articles
    -  use postman to test the search endpoint below:-
        - `http://127.0.0.1:8000/api/articles/?author=jake`
        - `http://127.0.0.1:8000/api/articles/?keyword=soccer`
        - `http://127.0.0.1:8000/api/articles/?tag=AngularJS`
        - `http://127.0.0.1:8000/api/articles/?favorited=joel`
#### Any background context you want to provide?
- N/A
#### What are the relevant pivotal tracker stories?
- [#164069234](https://www.pivotaltracker.com/story/show/164069234)

#### Screenshots (if appropriate)
* search by keyword
<img width="1138" alt="Screenshot 2019-03-18 at 16 10 09" src="https://user-images.githubusercontent.com/19649761/54532313-87942380-4998-11e9-80e1-7fff5bb2cd30.png">

* search by tag name
<img width="1138" alt="Screenshot 2019-03-18 at 16 10 44" src="https://user-images.githubusercontent.com/19649761/54532327-91b62200-4998-11e9-96bd-4781e7cca47e.png">
